server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -247,44 +247,48 @@ public static void splitTablet(KeyExtent extent, Text oldPrevEndRow, double spli
     update(context, zooLock, m, extent);
   }
 
-  public static void finishSplit(Text metadataEntry, Map<FileRef,DataFileValue> datafileSizes,
-      List<FileRef> highDatafilesToRemove, final ServerContext context, ZooLock zooLock) {
+  public static void finishSplit(Text metadataEntry, Map<TabletFile,DataFileValue> datafileSizes,
+      List<TabletFile> highDatafilesToRemove, final ServerContext context, ZooLock zooLock) {
     Mutation m = new Mutation(metadataEntry);
     TabletsSection.TabletColumnFamily.SPLIT_RATIO_COLUMN.putDelete(m);
     TabletsSection.TabletColumnFamily.OLD_PREV_ROW_COLUMN.putDelete(m);
     ChoppedColumnFamily.CHOPPED_COLUMN.putDelete(m);
 
-    for (Entry<FileRef,DataFileValue> entry : datafileSizes.entrySet()) {
-      m.put(DataFileColumnFamily.NAME, entry.getKey().meta(), new Value(entry.getValue().encode()));
+    for (Entry<TabletFile,DataFileValue> entry : datafileSizes.entrySet()) {
+      m.put(DataFileColumnFamily.NAME, entry.getKey().getMetadataText(),
+          new Value(entry.getValue().encode()));
     }
 
-    for (FileRef pathToRemove : highDatafilesToRemove) {
-      m.putDelete(DataFileColumnFamily.NAME, pathToRemove.meta());
+    for (TabletFile pathToRemove : highDatafilesToRemove) {
+      m.putDelete(DataFileColumnFamily.NAME, pathToRemove.getMetadataText());
     }
 
     update(context, zooLock, m, new KeyExtent(metadataEntry, (Text) null));
   }
 
-  public static void finishSplit(KeyExtent extent, Map<FileRef,DataFileValue> datafileSizes,
-      List<FileRef> highDatafilesToRemove, ServerContext context, ZooLock zooLock) {
+  public static void finishSplit(KeyExtent extent, Map<TabletFile,DataFileValue> datafileSizes,
+      List<TabletFile> highDatafilesToRemove, ServerContext context, ZooLock zooLock) {
     finishSplit(extent.getMetadataEntry(), datafileSizes, highDatafilesToRemove, context, zooLock);
   }
 
-  public static void addDeleteEntries(KeyExtent extent, Set<FileRef> datafilesToDelete,
+  /**
+   * datafilesToDelete are strings because they can be a TabletFile or directory
+   */
+  public static void addDeleteEntries(KeyExtent extent, Set<String> datafilesToDelete,
```

Comment:
Why did this end up `Set<String>`?

Reply:
This method was bumping into the GC code that uses tabletDirs.  One possible code path for a file to be added to those datafilesToDelete comes from ```GcVolumeUtil.getDeleteTabletOnAllVolumesUri()```  [Here](https://github.com/apache/accumulo/blob/8ea08f8ef59cd199ec23d414d8dbc89be7eb1667/server/master/src/main/java/org/apache/accumulo/master/TabletGroupWatcher.java#L627)

Reply:
Thanks for the pointer.